### DISCUSSION TITLE
No return value for signUp()/getUser()/getSession()/signInWithPassword()

### DISCUSSION THREAD

**Discussion by Epikur86** [supabase/supabase#9564] [Questions]
Hello,

I'm not able to get any useful return value from the function mentioned in the title. A user is created but I don't get any unique id for the user, in the console the user is only undefined. But the user is properly created in the database. It's the same for the other functions, return value is undefined. Any idea what I'm doing wrong?

```
async function signUpWithSupabase() {
    const { user, error } = await supabase.auth.signUp({
      email: username,
      password: password,
    });
    console.log("signup",user)

}
```

**Comment by GaryAustin1** [comment]
You use data instead of user in v2.
From the [docs](https://supabase.com/docs/reference/javascript/auth-signup):
```
const { data, error } = await supabase.auth.signUp({
  email: 'example@email.com',
  password: 'example-password',
})
```

**Reply by Epikur86** [reply]
The error is still the same. In the console the post request goes to this url "https://secret.supabase.co/auth/v1/signup" shouldn't it be v2?

**Reply by GaryAustin1** [reply]
The v1 in the URLs is the API server side version.  Supabase-js version can be seen in one of the headers or the supabase client object.  But you would get a function error if you were running v1 supabase-js and doing the other 3 calls.

**Reply by Epikur86** [reply]
Issue is solved, but I don't know how. Just tried several things, but cannot say what solved the issue.
